packages/console/src/hooks/use-save-redirect.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react';
+
+import { storageKeys } from '@/consts';
+
+/**
+ * A hook that return a function to save the current location to session storage
+ * for later redirect after sign-in.
+ */
+const useSaveRedirect = () => {
+  return useCallback(() => {
+    const { pathname, search, hash } = window.location;
+    sessionStorage.setItem(
+      storageKeys.redirectAfterSignIn,
+      JSON.stringify({ pathname, search, hash })
+    );
+  }, []);
+};
```

Comment:
Can we make it as a util function? Looks like no need for a hook.

Reply:
you are right, i was dumb